Index: b/frontends/spark_dvbapi5/stv090x.c
===================================================================
--- a/frontends/spark_dvbapi5/stv090x.c
+++ b/frontends/spark_dvbapi5/stv090x.c
@@ -6520,8 +6520,8 @@ static struct dvb_frontend_ops stv090x_o
 		.symbol_rate_max    = 70000000,
 		.caps           = FE_CAN_INVERSION_AUTO |
 		FE_CAN_FEC_AUTO       |
-		FE_CAN_QPSK           /*|
-                      FE_CAN_2G_MODULATION*/
+		FE_CAN_QPSK           |
+                      FE_CAN_2G_MODULATION
 	},
 
 	.release            = stv090x_release,
Index: b/frontends/spark7162/D3501.c
===================================================================
--- a/frontends/spark7162/D3501.c
+++ b/frontends/spark7162/D3501.c
@@ -5585,7 +5585,8 @@ static struct dvb_frontend_ops spark_d35
 		.symbol_rate_max    = 45000000,
 		.caps           = FE_CAN_INVERSION_AUTO |
 		FE_CAN_FEC_AUTO       |
-		FE_CAN_QPSK
+		FE_CAN_QPSK |
+		FE_CAN_2G_MODULATION
 	},
 
 	.init               = d3501_init,
