Index: b/frontends/spark_dvbapi5/stv090x.c
===================================================================
--- a/frontends/spark_dvbapi5/stv090x.c
+++ b/frontends/spark_dvbapi5/stv090x.c
@@ -5003,7 +5003,7 @@ static int stv090x_read_cnr(struct dvb_f
 		lock_f = STV090x_GETFIELD_Px(reg, LOCK_DEFINITIF_FIELD);
 		if (lock_f) {
 			msleep(5);
-			for (i = 0; i < 16; i++) {
+			for (i = 0; i < 4; i++) {
 				reg_1 = STV090x_READ_DEMOD(state, NNOSPLHT1);
 				val_1 = STV090x_GETFIELD_Px(reg_1, NOSPLHT_NORMED_FIELD);
 				reg_0 = STV090x_READ_DEMOD(state, NNOSPLHT0);
@@ -5011,7 +5011,7 @@ static int stv090x_read_cnr(struct dvb_f
 				val  += MAKEWORD16(val_1, val_0);
 				msleep(1);
 			}
-			val /= 16;
+			val /= 4;
 			snr = stv090x_table_lookup(stv090x_s2cn_tab, ARRAY_SIZE(stv090x_s2cn_tab) - 1, val);
 			div = stv090x_s2cn_tab[0].read - stv090x_s2cn_tab[ARRAY_SIZE(stv090x_s2cn_tab) - 1].read;
 			*cnr = 0xFFFF - ((val * 0xFFFF) / div);
@@ -5025,7 +5025,7 @@ static int stv090x_read_cnr(struct dvb_f
 		lock_f = STV090x_GETFIELD_Px(reg, LOCK_DEFINITIF_FIELD);
 		if (lock_f) {
 			msleep(5);
-			for (i = 0; i < 16; i++) {
+			for (i = 0; i < 4; i++) {
 				reg_1 = STV090x_READ_DEMOD(state, NOSDATAT1);
 				val_1 = STV090x_GETFIELD_Px(reg_1, NOSDATAT_UNNORMED_FIELD);
 				reg_0 = STV090x_READ_DEMOD(state, NOSDATAT0);
@@ -5033,7 +5033,7 @@ static int stv090x_read_cnr(struct dvb_f
 				val  += MAKEWORD16(val_1, val_0);
 				msleep(1);
 			}
-			val /= 16;
+			val /= 4;
 			snr = stv090x_table_lookup(stv090x_s1cn_tab, ARRAY_SIZE(stv090x_s1cn_tab) - 1, val);
 			div = stv090x_s1cn_tab[0].read - stv090x_s1cn_tab[ARRAY_SIZE(stv090x_s1cn_tab) - 1].read;
 			*cnr = 0xFFFF - ((val * 0xFFFF) / div);
